guides/rules/no-set-in-init.md

```diff
@@ -0,0 +1,5 @@
+# No this.set in `init`
+
+**TL;DR: Do not `this.set` inside the `init` lifecycle hook**
+
+`this.set` will register a property change with the object, but as `init` is the constructor, doing that work is unnecessary.
```

Comment:
I'd like a bit more context here. Is this true for all assignments that occur in `init`? If so, I think there is something to be said about consistency. For example, given the following:

```js
Component.extend({
  init() {
    this.name = 'foo';
  },
  actions: {
    changeName() {
      this.set('name', 'bar');
    }
  }
});
```

It is confusing and initially unclear why `set` was used in one situation but not the other. If the entire point of this is to avoid the cost of `set` during `init`, then it seems like that should probably be improved at the framework level.